backend/src/routes/courses.js

```diff
@@ -2,28 +2,46 @@ const express = require('express');
 const router = express.Router();
 const db = require('../db');
 const datefns = require('date-fns');
+const utils = require('../utils');
 
-router.get('/', async (req, res) => {
-    const timestampToDate = (date) => datefns.parse(Number(date), 'MM-DD-YYYY');
-    const toFormattedDate = (date) => datefns.format(date);
+const getCourses = async (req, res) => {
+    try {
+        const timestampToDate = (date) =>
+            datefns.parse(Number(date), 'MM-DD-YYYY');
+        const toFormattedDate = (date) => datefns.format(date);
 
-    // Default start and end date for the last week
-    const startDate = toFormattedDate(
-        req.query.startDate ? timestampToDate(req.query.startDate) : new Date()
-    );
-    const endDate = toFormattedDate(
-        req.query.endDate
-            ? timestampToDate(req.query.endDate)
-            : datefns.addWeeks(new Date(), 1)
-    );
+        // Default start and end date for the last week
+        const startDate = toFormattedDate(
+            req.query.startDate
+                ? timestampToDate(req.query.startDate)
+                : new Date()
+        );
+        const endDate = toFormattedDate(
+            req.query.endDate
+                ? timestampToDate(req.query.endDate)
+                : datefns.addWeeks(new Date(), 1)
+        );
 
-    const courses = await db.courses.getCourses(startDate, endDate);
-    const response = db.courses.reduceCoursesByDate(courses);
-    res.status(200).send(response);
-});
+        const courses = await db.courses.getCourses(startDate, endDate);
+        const response = db.courses.reduceCoursesByDate(courses);
+        res.status(200).send(response);
+    } catch (err) {
+        res.status(420).send('Failed to get courses: ' + err.message);
+    }
+};
 
-router.get('/:id', (req, res) => {
-    res.send('returning singe course ' + req.params.id);
+const getCourseById = (utils.courses.validateCourseId,
```

Comment:
It would be more clean to mark the middleware in the route definitions like `router.get(utils.courses.validateCourseId, getCourseById)`. This way we can see which middlewares are applied for any route.